### PR TITLE
Make erroneous / dangerous links display as text only

### PR DESCRIPTION
**Pull request by jeremyblalock**
**TL;DR:** Leaves link text but removes hyperlink.

## Explanation

Currently when there is a link that either cannot be parsed, or is detected to be potentially malicious, **marked** simply hides the entire link. For example, if I had the following markdown:

```
For help: [Click Here](data:text/html,lkjasdjflkasd==).
```

the output would be the following:

> For help: .

_This is especially problematic_ when URLs are automatically converted to links. For example:

```
http://example.com
```

Would automatically become:

> http://example.com

If for some reason the URL cannot be parsed, then the URL simply does not display at all. This happens if it has un-urlencoded characters etc., or if it otherwise cannot be parsed, but _was_ identified as a URL by the separate lexer.

### CODE REVIEW

**Comment by matt-** [comment]
I like this idea. It looks like a simple change as well, but do you mind adding a test?

**Comment by jeremyblalock** [comment]
@matt- Yeah, I can add a test for it.

**Comment by jeremyblalock** [comment]
@matt- I updated the invalid link test / added a new one for the `decodeURIComponent` failure. 

On another note, noticed that we are not quite following the recommended test setup as described in the documentation, though. I don't see `links.sanitize.html` in `tests/new`. And so when you run `node test --fix` it will delete this test case, which I'm guessing is not the desired behavior. Are we ditching the `test/original` / `test/new` structure in favor of just putting everything in `/test/tests`? Let me know, I can move the files around if necessary get the `node test --fix` working properly.

**Comment by jeremyblalock** [comment]
@matt- @chjj: any decision on this? This will produce the same behavior that github has in their markdown rendering support for data: urls.

**Comment by egeste** [comment]
+1

**Comment by Insayt** [comment]
<ul><li>123123123</li></ul>

**Comment by Insayt** [comment]
# Hello world

**Comment by Insayt** [comment]
# 

> ``

**Comment by Feder1co5oave** [comment]
This looks good to me. +1
